conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-testing.gradle.kts

```diff
@@ -106,22 +106,24 @@ afterEvaluate {
     // We do fine-grained filtering of the classpath of this codebase's sources since Gradle's
     // configurations will include transitive dependencies as well, which tests do often need.
     classpath = classpath.filter {
-      // The sources are packaged into the testing jar so we need to make sure to exclude from the test
-      // classpath, which automatically inherits them, to ensure our shaded versions are used.
       if (file("${buildDir}/resources/main").equals(it) || file("${buildDir}/classes/java/main").equals(it)) {
+        // The sources are packaged into the testing jar, so we need to exclude them from the test
+        // classpath, which automatically inherits them, to ensure our shaded versions are used.
         return@filter false
       }
 
-      // TODO(anuraaga): Better not to have this folder structure constraints, we can likely use
+      // TODO(anuraaga): Better not to have this naming constraint, we can likely use
       // plugin identification instead.
 
-      // If agent depends on some shared instrumentation module that is not a testing module, it will
-      // be packaged into the testing jar so we need to make sure to exclude from the test classpath.
       val lib = it.absoluteFile
-      val instrumentationDir = file("${rootDir}/instrumentation/").absoluteFile
-      if (lib.startsWith(instrumentationDir) &&
-        lib.extension == "jar" &&
-        !lib.absolutePath.substring(instrumentationDir.absolutePath.length).contains("testing")) {
+      if (lib.name.startsWith("opentelemetry-javaagent-")) {
```

Comment:
If `lib` is an `absoluteFile`, how can it start with `opentelemetry-javaagent`?

Reply:
it's `lib.name`, e.g. `File.getName()`, not the full path